src/home/sorting/sorting-manager.ts

```diff
@@ -131,31 +132,50 @@ export class SortingManager {
   }
 
   private _handleSortingClick(input: HTMLInputElement, option: string) {
-    const ordering = input === this._lastChecked ? "reverse" : "normal";
-    input.checked = input !== this._lastChecked;
-    input.setAttribute("data-ordering", ordering);
-    this._lastChecked = input.checked ? input : null;
-    this._filterTextBox.value = "";
+    const currentOrdering = input.getAttribute("data-ordering");
+    let newOrdering: string;
+
+    // Determine the new ordering based on the current state
+    if (currentOrdering === "normal") {
+      newOrdering = "reverse";
+    } else if (currentOrdering === "reverse") {
+      newOrdering = "disabled";
+    } else {
+      newOrdering = "normal";
+    }
+
+    // Apply the new ordering state
+    input.setAttribute("data-ordering", newOrdering);
     input.parentElement?.childNodes.forEach((node) => {
       if (node instanceof HTMLInputElement) {
         node.setAttribute("data-ordering", "");
       }
     });
 
-    input.setAttribute("data-ordering", ordering);
-    // instantly load from cache
+    if (newOrdering === "disabled") {
+      this._lastChecked = null;
+      input.checked = false;
+      this._clearSorting();
+    } else {
+      input.checked = input !== this._lastChecked;
+      this._lastChecked = input.checked ? input : null;
+      input.setAttribute("data-ordering", newOrdering);
+
+      // Apply the sorting based on the new state (normal or reverse)
+      try {
+        void displayGitHubIssues(option as Sorting, { ordering: newOrdering });
+      } catch (error) {
+        renderErrorCatch(error as ErrorEvent);
+      }
+    }
+  }
+
+  private _clearSorting() {
     try {
-      void displayGitHubIssues(option as Sorting, { ordering });
+      void displayGitHubIssues();
     } catch (error) {
-      renderErrorCatch(error as ErrorEvent);
+      renderErrorInModal(error as Error);
     }
-    // load from network in the background
-    // const fetchedPreviews = await fetchIssuePreviews();
-    // const cachedTasks = taskManager.getTasks();
-    // const updatedCachedIssues = verifyGitHubIssueState(cachedTasks, fetchedPreviews);
-    // displayGitHubIssues(sorting, options);
-    // taskManager.syncTasks(updatedCachedIssues);
-    // return fetchAvatars();
```

Comment:
Are you going to delete?

Reply:
This was old code, can remove that safely